pynetgear/__main__.py

```diff
@@ -1,22 +1,101 @@
 """Run PyNetgear from the command-line."""
 import sys
+import os
+
+from argparse import ArgumentParser
 from pynetgear import Netgear
 
 
+def make_formatter(format_name):
+    """Returns a callable that outputs the data. Defaults to csv."""
+
+    if format_name == "json":
+        from json import dumps
+
+        def jsonify(devices):
+            print(dumps([device._asdict() for device in devices]))
+        return jsonify
+    else:
+        return (lambda devices: [print(device) for device in devices])
+
+
+def argparser():
+    parser = ArgumentParser(prog='pynetgear')
+
+    parser.add_argument("--format", choices=['json', 'py'], default='py')
+    # parser.add_argument("--router-config", type=FileType('r'), help="Read router config from file.")
+
+    router_args = parser.add_argument_group("router connection config")
+    router_args.add_argument("--ssl", action="store_true", help="Connect with https")
+    router_args.add_argument("--host", help="Hostname for the router")
+    router_args.add_argument("--user", help="Account for login")
+    router_args.add_argument("--port", help="Port exposed on the router")
+    router_args.add_argument("--url", help="....")
+
+    subparsers = parser.add_subparsers(description="Runs subcommand against the specified router", dest="subcommand")
+
+    block_parser = subparsers.add_parser("block_device", help="Blocks a device from connecting by mac address")
+    block_parser.add_argument("--mac-addr")
+
+    allow_parser = subparsers.add_parser("allow_device", help="Allows a device with the mac address to connect")
+    allow_parser.add_argument("--mac-addr")
+
+    subparsers.add_parser("login", help="Attempts to login to router.")
+
+    attached_devices = subparsers.add_parser("attached_devices", help="Outputs all attached devices")
+    attached_devices.add_argument(
+            "-v", "--verbose",
+            action="store_true",
+            default=False,
+            help="Choose between verbose and slower or terse and fast.")
+
+    subparsers.add_parser("traffic_meter", help="Output router's traffic meter data")
+
+    # parser.add_argument("-", dest="router_config", help="PASSWORD HOST USER PORT SSL")
+    # parser.add_argument(" -- ", nargs="*", help="[<pass>] [<host>] [<user>] [<port>] [<ssl>]")
```

Comment:
I wanted this to have a `--` seperator to fallback to the old behavior for minimally invasive change but I had difficulty getting that to work within the `ArgumentParser` and providing the CLI help documentation.